sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java

```diff
@@ -208,9 +212,12 @@ public synchronized void removeEventProperty(String key) {
         mEventProperties.remove(key);
     }
 
-    @SuppressLint("HardwareIds")
+    /**
+     * Enable collection of the Android device identifier for this target.
+     * This does not have any effect on child transmission targets.
+     */
     public void collectDeviceId() {
-        mDeviceId = Secure.getString(mTransmissionTarget.mContext.getContentResolver(), Secure.ANDROID_ID);
+        mDeviceIdEnabled = true;
```

Comment:
Context can be null when calling this early so we just remember the call and apply later.

Reply:
Understand this change but we aligned the iOS implementation matching with Android.
Not a big deal but we need to change it in iOS, too.

Reply:
I don't think we should change iOS as this constraint is a bug fix for Android.